### PR TITLE
Allow to update che server deploymentConfig

### PR DESCRIPTION
**Pull request by riuvshin**
Allow to update che server deploymentConfig

here I replaced `oc rollout latest che` with `oc apply` as we have 
```
    triggers:
    - type: ConfigChange
```
But I'm not 100% sure that it is a equivalent so **dear** @reviewers if you have any concerns about that please feel free to share your opinion.

fixes: https://github.com/eclipse/che/issues/7704

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh [hunk 1]
```diff
@@ -275,38 +287,14 @@ echo -n "[CHE] Switching to \"${CHE_OPENSHIFT_PROJECT}\"..."
...
-
 # -------------------------------------------------------------
 # Deploying secondary servers
 # for postgres and optionally Keycloak
 # -------------------------------------------------------------
 
 COMMAND_DIR=$(dirname "$0")
 
-if [ "${CHE_MULTIUSER}" == "true" ]; then
+if [[ "${CHE_MULTIUSER}" == "true" ]] && [[ "${COMMAND}" != *"rollupdate"* ]]; then
```
Comment (sleshchenko) [inline dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh]: Will be this statement true if the command is `cleanup`?

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh [hunk 2]
```diff
@@ -344,6 +332,29 @@ else
   CHE_KEYCLOAK_CLIENT__ID=${CHE_KEYCLOAK_CLIENT__ID:-"openshiftio-public"}
 fi
 
+# -------------------------------------------------------------
+# If command == cleanup then delete all openshift objects
+# -------------------------------------------------------------
+if [ "${COMMAND}" == "cleanup" ]; then
```
Comment (sleshchenko) [inline dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:338]: Can you explain why did you move these lines?

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh (hunk 1 above)
Comment (riuvshin) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh]: good catch Sergio! Fixed! Thanks

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh (hunk 2 above)
Comment (riuvshin) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:338]: Thanks for the question Sergio! By my nature I love moving lines around even without any reasons, but not in that case... Here I moved them (with a pleasure I must admit) because in case of `rollupdate ` we call `get_che_pod_config | oc apply -f -` and here before we applied configuration update we need all ENV vars used in `DEFAULT_CHE_CONFIG_FILE_PATH=./che-config` to be inited. And as you can see here before my changes these lines were before `# Setting Keycloak-related environment variables
` section where we init KC related ENVs so that's why.

**Review by sleshchenko — APPROVED** [review]

File: dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh (hunk 2 above)
Comment (sleshchenko) [reply dockerfiles/init/modules/openshift/files/scripts/deploy_che.sh:338]: Thanks for the explanation!

**Review by garagatyi — APPROVED** [review]
